src/inmanta/data/__init__.py

```diff
@@ -5533,6 +5537,21 @@ class EnvironmentMetricsTimer(BaseDocument):
     __primary_key__ = ("environment", "metric_name", "category", "timestamp")
 
 
+class User(BaseDocument):
+    """A user that can authenticate against inmanta"""
+
+    __primary_key__ = ("id",)
+
+    id: uuid.UUID
+    username: str
+    password_hash: str
+    auth_method: str
```

Comment:
I tried to change that but I get an issue with the insert:
```
        try:
            user = data.User(
                username=username,
                password_hash=pw_hash.decode(),
                auth_method=AuthMethod.DATABASE,
            )
            await user.insert()
```
It tries to insert the key of the enum ('DATABASE') instead of the value ('database') which isn't a valid value in the DB.

Reply:
I see that all other enums used in DAO classes resolve this by using small letter in the key of the enum instead of capital letters. This goes against the Python convention of using capital letters for the key of an enum, but I think that the benefits of improved readability by using the enum instead of the str type outweighs the downside of violating that naming convention.

Reply:
fixed :)

Reply:
> Looks like this has been there forever (since the migration to Postgres). @arnaudsjs do you know why?

I missed this message. Probably as issue with by browser cache. I don't recall why this was required, but probably asyncpg doesn't know what to do if the type in the database is an enum and you pass in a enum as a parameter to the query.

Reply:
Indeed.